src/ray/gcs/gcs_server/gcs_placement_group_manager.cc

```diff
@@ -154,29 +152,31 @@ void GcsPlacementGroupManager::HandleCreatePlacementGroup(
     ray::rpc::SendReplyCallback send_reply_callback) {
   auto placement_group_id =
       PlacementGroupID::FromBinary(request.placement_group_spec().placement_group_id());
-  const auto &strategy = request.placement_group_spec().strategy();
   const auto &name = request.placement_group_spec().name();
+  const auto &strategy = request.placement_group_spec().strategy();
   RAY_LOG(INFO) << "Registering placement group, placement group id = "
                 << placement_group_id << ", name = " << name
                 << ", strategy = " << PlacementStrategy_Name(strategy);
-  RegisterPlacementGroup(
-      request, [reply, send_reply_callback, placement_group_id](
-                   std::shared_ptr<gcs::GcsPlacementGroup> placement_group) {
-        RAY_LOG(INFO) << "Registered placement group, placement group id = "
-                      << placement_group_id << ", name = " << placement_group->GetName()
-                      << ", strategy = " << placement_group->GetStrategy();
-        GCS_RPC_SEND_REPLY(send_reply_callback, reply, Status::OK());
-      });
   auto placement_group = std::make_shared<GcsPlacementGroup>(request);
-  auto placement_group_table_data = placement_group->GetPlacementGroupTableData();
   RAY_CHECK_OK(gcs_table_storage_->PlacementGroupTable().Put(
-      placement_group_id, placement_group_table_data, [](Status status) {}));
+      placement_group_id, placement_group->GetPlacementGroupTableData(),
+      [this, request, reply, send_reply_callback, placement_group_id, name,
+       strategy](Status status) {
+        RAY_CHECK_OK(status);
+        RegisterPlacementGroup(request, [reply, send_reply_callback, placement_group_id,
+                                         name, strategy]() {
+          RAY_LOG(INFO) << "Finished registering placement group, placement group id = "
+                        << placement_group_id << ", name = " << name
+                        << ", strategy = " << strategy;
+          GCS_RPC_SEND_REPLY(send_reply_callback, reply, Status::OK());
+        });
+      }));
 }
 
 void GcsPlacementGroupManager::ScheduleTick() {
   reschedule_timer_.expires_from_now(boost::posix_time::milliseconds(500));
   reschedule_timer_.async_wait([this](const boost::system::error_code &error) {
-    if (error == boost::system::errc::operation_canceled) {
+    if (error == boost::asio::error::operation_aborted) {
```

Comment:
Just out of curiosity. Are they different error code, or same?

Reply:
The value of boost::system::errc::operation_canceled and boost::asio::error::operation_aborted is same. But It is recommended to use boost::asio::error::operation_aborted in the document of deadline_timer.